praxis-isncsci/ui
# App bar

Closes #72 

## Problem

The application needs an app bar that allows the use of the following optional subcomponents:

- Menu button (hamburger icon button)
- Title
- Actions (buttons like "calculate", "download", etc.)

## Solution

We are favoring composition so we add slots for the optional subcomponents.
This will make it easy for the component or controller adding them to control their appearance and subscribe to their events.

## Testing

- [x] 1. Can create an app bar that loads a menu button, title, and actions

<details>
  <summary>Test case</summary>

  1. Navigate to the story `?path=/docs/webcomponents-app-bar--docs`
  2. Confirm  that we can display an app bar with the following optional elements
      - Menu button
      - Title
      - Actions

</details>
